### PR TITLE
Adding readwrite-primary mode

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
This exposes three modes for an IndexedDbTransaction (readonly, readwrite, readwrite-primary) and removes the `requiresPrimaryLease` flag.

### CODE REVIEW

**Comment by schmidt-sebastian** [comment]
Woohoo. 913 lines.

**Comment by mikelehen** [comment]
btw, you didn't remove the actual TODO. 😛

File: packages/firestore/src/local/indexeddb_persistence.ts [hunk 1]
```diff
@@ -724,10 +728,10 @@ export class IndexedDbPersistence implements Persistence {
     // Do all transactions as readwrite against all object stores, since we
     // are the only reader/writer.
     return this.simpleDb.runTransaction(
-      'readwrite',
+      mode == 'readonly' ? 'readonly' : 'readwrite',
```
Comment (mikelehen) [inline packages/firestore/src/local/indexeddb_persistence.ts]: If you end up touching this PR before submitting (e.g. to remove the TODO), change this to `===`?  Else it's not really important.

Comment (schmidt-sebastian) [reply packages/firestore/src/local/indexeddb_persistence.ts]: Done.

**Comment by schmidt-sebastian** [comment]
> btw, you didn't remove the actual TODO. 😛

I'll remove it in the other PR.

**Pull request merged**